File: Data Structures and Algorithms/LeetCode/169. Majority Element.cs
public class Solution {
    public int MajorityElement(int[] nums) {
        
        int majority=nums.Length/2;
        Console.WriteLine(majority);
        int b=0;
        
        int[] array =new int[nums.Length];
        Array.Sort(nums);
         
        int count=1;
        
        for(int i=0;i<nums.Length-1;i++){
        if( nums[i] ==nums[nums.Length-1]){
            count=count+1;
            
            if(count>majority) return nums[i];
        }         
            if(nums[i]==nums[i+1]){
                count=count+1;   
            }
            else 
            {
                if(count>majority)  {return nums[i]; count=1;b++;}
            }
        }
        
      if(nums.Length==1) return nums[0];
        return -100;
    }
}

File: Data Structures and Algorithms/LeetCode/169. Majority Element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        # boyer moore algo

        count = 0
        result = 0

        for n in nums:
            if count == 0:
                result = n
            if n == result:
                count += 1
            else:
                count -= 1
        return result


class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        hashmap = {}

        for i in range(len(nums)):
            hashmap[nums[i]] = 1 + hashmap.get(nums[i], 0)
        return max(hashmap, key=hashmap.get)
